golang/internal/helper/docker/container.go

```diff
@@ -27,16 +27,33 @@ func DeleteContainerByName(ctx context.Context, dog *dogger.DeploymentLogger, na
 	return deleteContainerByState(ctx, dog, nameFilter, matchedContainer.State, errorReport)
 }
 
-func DeleteContainerByID(ctx context.Context, dog *dogger.DeploymentLogger, nameFilter string, errorReport bool) error {
-	matchedContainer, err := GetContainerByID(ctx, nil, nameFilter, errorReport)
+func DeleteContainerByID(ctx context.Context, dog *dogger.DeploymentLogger, idFilter string, errorReport bool) error {
+	matchedContainer, err := GetContainerByID(ctx, nil, idFilter, errorReport)
 	if err != nil {
-		return fmt.Errorf("builder could not get container (%s) to remove: %s", nameFilter, err.Error())
+		return fmt.Errorf("builder could not get container (%s) to remove: %s", idFilter, err.Error())
 	}
 	if matchedContainer == nil {
 		return nil
 	}
 
-	return deleteContainerByState(ctx, dog, nameFilter, matchedContainer.State, errorReport)
+	return deleteContainerByState(ctx, dog, idFilter, matchedContainer.State, errorReport)
+}
+
+func DeletePrefix(ctx context.Context, prefix string) error {
+	containers, err := GetAllContainersByName(ctx, nil, prefix)
+
+	if err != nil {
+		return fmt.Errorf("builder could not get containers for prefix (%s) to remove: %s", prefix, err.Error())
```

Comment:
we are not in the builder.

Reply:
i've just copied this from above, probably that's faulty as well?